lxl-web/src/routes/(app)/[[lang=lang]]/(find)/+layout.server.ts

```diff
@@ -0,0 +1,136 @@
+import { redirect, error } from '@sveltejs/kit';
+import jmespath from 'jmespath';
+import { env } from '$env/dynamic/private';
+import { getSupportedLocale } from '$lib/i18n/locales.js';
+import { getTranslator } from '$lib/i18n/index.js';
+import { type FramedData, DisplayUtil, pickProperty } from '$lib/utils/xl.js';
+import { LxlLens } from '$lib/utils/display.types.js';
+import { getImageLinks, generateAuxdImageUri, calculateExpirationTime } from '$lib/utils/auxd';
+import addDefaultSearchParams from '$lib/utils/addDefaultSearchParams.js';
+import { type apiError } from '$lib/types/API.js';
+import { asResult, type PartialCollectionView, type SearchResult } from './search.js';
+
+export const load = async ({ params, url, locals, fetch, isDataRequest }) => {
+	const displayUtil: DisplayUtil = locals.display;
+	const locale = getSupportedLocale(params?.lang);
+
+	const isResourceRoute = !!params.fnurgel;
+	const isFindRoute = url.pathname === '/find';
+	let resourceParts = {};
+	let searchResult: SearchResult | null = null;
+	let shouldFindRelations, resourceId;
+
+	// If product page, get the resource
+	if (isResourceRoute) {
+		const resourceRes = await fetch(`${env.API_URL}/${params.fnurgel}?framed=true`, {
+			headers: { Accept: 'application/ld+json' }
+		});
+
+		if (!resourceRes.ok) {
+			const err = (await resourceRes.json()) as apiError;
+			// Todo better error handling
+			throw error(err.status_code, err.status);
+		}
+
+		const resource = await resourceRes.json();
+		const mainEntity = centerOnWork(resource['mainEntity'] as FramedData);
+		resourceId = resource.mainEntity['@id']; // can we rely on this???
```

Comment:
Getting the id for the `o` search like this, is it safe?

Reply:
Yes!